### PR TITLE
Remove unused React component generator

### PR DESCRIPTION
**Pull request by thibaudcolas**
Initially discussed removing this over at https://github.com/wagtail/wagtail/pull/4709#pullrequestreview-142143097.

I think this generator was added at the same time as the Webpack tooling, for the new explorer menu. The scaffolding might have worked but we've never used it much simply because we don't create new components that often. As far as I know, this was not documented anywhere anyway (beyond having the `npm run component` command), and removing it shouldn't break anything.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,7 +76,6 @@
     "gulp-sourcemaps": "~2.6.1",
     "gulp-util": "~3.0.8",
     "jest": "^22.0.3",
-    "mustache": "^2.2.1",
```
Comment (thibaudcolas) [inline package.json]: I'm not sure what this was doing here. Removing it isn't related to the rest of the changes, but as far as I can see this dependency isn't used anywhere (anymore?).

Comment (BertrandBordage) [reply package.json]: Well, mustache was used to generate the .mst template, right?
So it’s in fact related to the other changes you did :)

Comment (thibaudcolas) [reply package.json]: Well true :) What I meant was that the `mustache` used for this generator was declared in `client/package.json` – so there shouldn't have been one for that purpose in the root package.json.

**Comment by gasman** [comment]
Merged in ad5a3bbdc2ee9a3c9adb3150ed71eaa403a8390e / 6301eb4b75c7829bee33b334ef6ef0a755bd3dbd

**Pull request closed without merging**